evm_loader/program/src/instruction/collect_treasury.rs

```diff
@@ -0,0 +1,28 @@
+use crate::account::{Treasury, MainTreasury};
+use arrayref::{array_ref};
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar::Sysvar,
+    rent::Rent,
+};
+
+pub fn process<'a>(program_id: &'a Pubkey, accounts: &'a [AccountInfo<'a>], instruction: &[u8]) -> ProgramResult {
+    solana_program::msg!("Instruction: Collect treasury");
+
+    let treasury_index = u32::from_le_bytes(*array_ref![instruction, 0, 4]);
+
+    let main_treasury = MainTreasury::from_account(&accounts[0])?;
+    let treasury = Treasury::from_account(program_id, treasury_index, &accounts[1])?;
+
+    let rent = Rent::get()?;
+    let minimal_balance_for_rent_exempt = rent.minimum_balance(treasury.data_len());
+    let available_lamports = treasury.lamports().saturating_sub(minimal_balance_for_rent_exempt);
+
+    **treasury.lamports.borrow_mut() = minimal_balance_for_rent_exempt;
+    **main_treasury.lamports.borrow_mut() = main_treasury.lamports().checked_add(available_lamports)
```

Comment:
In addition, to reflect changes in the spl-token balance you need to send [sync_native](https://docs.rs/spl-token/latest/spl_token/instruction/fn.sync_native.html) instruction to the spl-token program.

Reply:
Executed by the last command in `neon-cli collect-treasury`. When I try to do this inside the neon-evm instruction, I get an error `UnbalancedInstruction` (`Program DCPSnJHB38e7vNK6o3AVLswJGRaP87iiNx2zvvapiKBz failed: sum of account balances before and after instruction do not match`)